astropy/stats/funcs.py

```diff
@@ -681,15 +681,29 @@ def median_absolute_deviation(a, axis=None):
 
     """
 
-    a = np.array(a, copy=False)
-    a_median = np.median(a, axis=axis)
+    # Check if the array has a mask and if so use np.ma.median
+    # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
+    # for normal arrays should not be done (summary: np.ma.median always
+    # returns an masked array even if the result should be scalar).
+    try:
+        if np.any(a.mask):
```

Comment:
Why this whole check rather that just using `np.ma.median` for `isinstance(a, MaskedArray)`? I think if one passes in a masked array, the expectation should be to get a masked array back!